airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -80,6 +81,44 @@ def ensure_at_least_one_state_emitted(self) -> None:
         pass
 
 
+class FinalStateCursor(Cursor):
+    """Cursor that is used to guarantee at least one state message is emitted for a concurrent stream."""
+
+    def __init__(
+        self,
+        stream_name: str,
+        stream_namespace: Optional[str],
+        message_repository: MessageRepository,
+    ) -> None:
+        self._stream_name = stream_name
+        self._stream_namespace = stream_namespace
+        self._message_repository = message_repository
+        # Normally the connector state manager operates at the source-level. However, we only need it to write the sentinel
+        # state message rather than manage overall source state. This is also only temporary as we move to the resumable
+        # full refresh world where every stream uses a FileBasedConcurrentCursor with incremental state.
```

Comment:
The explanation for `FileBasedConcurrentCursor` looks fair. Does the same apply to `ConcurrentCursor`?

Reply:
yeah both, i can add it there too

Reply:
I don't know, maybe I got confused to see something file based specific on a file that shouldn't know about file based. That being said, the logic of instantiating it here makes sense to me